config/types/etcd.go

```diff
@@ -147,6 +150,47 @@ func etcdContents(etcd Etcd, platform string) (string, error) {
 	return unit.String(), nil
 }
 
+var validPrefixes = []string{"http://", "https://", "unix://", "unixs://"}
+
+func checkForPrefix(s string) bool {
+	if s == "" {
+		return true
+	}
+	urlStrs := strings.Split(s, ",")
+	for _, urlStr := range urlStrs {
+		hasPrefix := false
+		for _, p := range validPrefixes {
+			hasPrefix = hasPrefix || strings.HasPrefix(urlStr, p)
```

Comment:
I'm not familiar with the full URL RFC. You'll need to review that before this is safe to do. Otherwise, substitute a dummy IP address before validation.